base/linalg/symmetric.jl

```diff
@@ -41,6 +41,15 @@ julia> Slower = Symmetric(A, :L)
 Note that `Supper` will not be equal to `Slower` unless `A` is itself symmetric (e.g. if `A == A.'`).
 """
 Symmetric(A::AbstractMatrix, uplo::Symbol=:U) = (checksquare(A); Symmetric{eltype(A),typeof(A)}(A, char_uplo(uplo)))
+Symmetric(A::Symmetric) = A
+function Symmetric(A::Symmetric, uplo::Symbol)
+    if A.uplo == char_uplo(uplo)
+        return A
+    else
+        throw(ArgumentError("Cannot construct Symmetric; uplo doesn't match"))
```

Comment:
why not just ignore the uplo argument? Presumably, if you're building a symmetric matrxx from an already symmetric matrix, taking the lower or upper triangle should not matter since they are the same.